src/Analyser/NodeScopeResolver.php

```diff
@@ -4522,8 +4524,22 @@ private function processArgs(
 			}
 
 			if ($assignByReference) {
+				if ($currentParameter === null) {
+					throw new ShouldNotHappenException();
+				}
+
 				$argValue = $arg->value;
 				if ($argValue instanceof Variable && is_string($argValue->name)) {
```

Comment:
why does this code path only exec on variables and not e.g. on PropertyFetch?

Reply:
You can explore that in another PR.